geode-core/src/main/java/org/apache/geode/internal/cache/LocalRegion.java

```diff
@@ -5035,8 +5029,11 @@ public void txApplyPutHandleDidDestroy(Object key) {
   public boolean basicPut(EntryEventImpl event, boolean ifNew, boolean ifOld,
       Object expectedOldValue, boolean requireOldValue)
       throws TimeoutException, CacheWriterException {
-    return getDataView().putEntry(event, ifNew, ifOld, expectedOldValue, requireOldValue, 0L,
-        false);
+    boolean returnValue =
+        getDataView().putEntry(event, ifNew, ifOld, expectedOldValue, requireOldValue, 0L,
+            false);
+    getFutures.remove(event.getKey());
```

Comment:
Need to handle exception (in case of timeout exception, the changes may have been applied on server).
And remove only if its proxy?